### ISSUE TITLE
What does ppl mean?

### ISSUE THREAD

**Issue by mikiihuang** [AKSW/NSpM#11]
thanks for your excellent work in the interesting issue.
When i'm training with your given monument_300 data,I saw the output like this:

`step 4100 lr 1 step-time 2.35s wps 2.36K ppl 64.10 gN 3.08 bleu 2.74, Sat Dec  8 14:28:50 2018`
Can you tell me what does the ppl and gN mean? And why is the BLEU score so small?
Thank you very much.

**Comment by mommi84** [comment]
Hi @mikiihuang.

1. `ppl` means perplexity and `gN` means gradient norm.
2. How long did you train the model? It is normal to start with a low BLEU score. In the papers, we reported the elapsed runtime of our experiments.

Cheers!

**Comment by mikiihuang** [comment]
> Hi @mikiihuang.
> 
> 1. `ppl` means perplexity and `gN` means gradient norm.
> 2. How long did you train the model? It is normal to start with a low BLEU score. In the papers, we reported the elapsed runtime of our experiments.
> 
> Cheers!

Thank you very much!! Actually I didn't run it for a long time due to the memory issue.However the work you did really inspires me 👍Cheers.
